workflow/common/util.go

```diff
@@ -291,6 +292,22 @@ func addPodMetadata(c kubernetes.Interface, field, podName, namespace, key, valu
 	return err
 }
 
+const deleteRetries = 3
+
+// deletePod return error to indicate delete pod successfully or not.
+func DeletePod(c kubernetes.Interface, podName, namespace string) error {
+	var err error
+	for attempt := 0; attempt < deleteRetries; attempt++ {
+		err = c.CoreV1().Pods(namespace).Delete(podName, &metav1.DeleteOptions{})
+		if err != nil {
```

Comment:
I think the `apierr.IsNotFound(err)` check should be moved here, because this logic is retrying deletion errors due to NotFound pods.